### DISCUSSION TITLE
loading workflows from JSON and workflow variables

### DISCUSSION THREAD

**Discussion by mremadevi** [elsa-workflows/elsa-core#5297] [Q&A]
Can loading workflows from JSON and executing from code make use of workflow variables for custom activity output storage? How can next activity access the assigned variable?

**Comment by glime-ai** [comment]
Yes, loading workflows from JSON and executing them from code can make use of workflow variables for custom activity output storage. When a workflow is defined in JSON, you can specify output properties for activities, which can then be accessed by subsequent activities within the workflow.

To set the value of an output variable in a custom activity, you would typically use the `context.SetVariable("VariableName", value)` method within the custom activity's execution logic. This stores the value in the workflow's context under the specified variable name.

Subsequent activities can access this variable by using the `context.GetVariable<T>("VariableName")` method, where `T` is the expected type of the variable's value.

In the case of workflows defined in JSON, you can reference the output of a previous activity using expressions. For example, if you have a custom activity that sets an output variable, you can use the output in a `WriteLine` activity by specifying an expression that accesses the variable, such as `"{{ Variables.VariableName }}"` if using Liquid syntax or `${getVariableName()}` if using JavaScript syntax.

Here's an example of how you might define this in JSON:

```json
{
  "activities": [
    {
      "id": "customActivity",
      "type": "CustomActivityType",
      "state": {
        // Custom activity state
      }
    },
    {
      "id": "writeLineActivity",
      "type": "WriteLine",
      "state": {
        "textExpression": {
          "expression": "{{ Activities.customActivity.OutputVariableName }}",
          "syntax": "Liquid"
        }
      }
    }
  ],
  "connections": [
    {
      "sourceActivityId": "customActivity",
      "destinationActivityId": "writeLineActivity",
      "outcome": "Done"
    }
  ]
}
```

In this example, `CustomActivityType` is a placeholder for your custom activity's type name, and `OutputVariableName` is the name of the output variable set by the custom activity. The `WriteLine` activity then uses this variable in its `textExpression`.

Remember that the exact syntax for accessing variables may vary depending on the expression language used in the workflow definition (e.g., Liquid or JavaScript).

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
